modules/mod_ginger_base/controllers/controller_search.erl

```diff
@@ -102,4 +104,37 @@ whitelisted(Arguments) ->
 %% @doc Get whitelisted search arguments.
 -spec whitelist() -> [atom()].
 whitelist() ->
-    [cat, cat_promote_recent, filter, hasobject, hassubject, text, sort, has_geo].
+    [cat, cat_promote_recent, facet, filter, global_facet, hasobject, hassubject, text, sort, has_geo].
+
+%% @doc Combine separate facets (date_start_min, date_start_max) into one property
+%% (date_start.min, date_start.max).
+-spec facets(map()) -> map().
+facets(Facets) ->
+    maps:fold(
+        fun(K, V, Acc) ->
+            Skip = byte_size(K) - 4,
+            Seven = byte_size(K) - 7,
+            case K of
+                <<Field:Skip/binary, "_min">> ->
+                    Acc#{Field => maps:put(<<"min">>, V, maps:get(Field, Acc, #{}))};
+                <<Field:Skip/binary, "_max">> ->
+                    Acc#{Field => maps:put(<<"max">>, V, maps:get(Field, Acc, #{}))};
+                <<Field:Seven/binary, "_global">> ->
```

Comment:
@rmies These patterns are here to match `_min|_max` and `_global`, reactive to the output received from `z_search:search/3`. These patterns look somewhat messy. An alternative is to use the user-supplied list of facets (as query string param) as a starting point. Do you have any suggestions to clean this up?

Reply:
I'm afraid not.